presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsCalculator.java

```diff
@@ -195,18 +197,31 @@ private PlanNodeStatsEstimate getStatistics(PlanNode planNode, Session session,
         for (PlanCanonicalizationStrategy strategy : historyBasedPlanCanonicalizationStrategyList(session)) {
             for (Map.Entry<PlanNodeWithHash, HistoricalPlanStatistics> entry : statistics.entrySet()) {
                 if (allHashes.containsKey(strategy) && entry.getKey().getHash().isPresent() && allHashes.get(strategy).equals(entry.getKey())) {
-                    Optional<List<PlanStatistics>> inputTableStatistics = getPlanNodeInputTableStatistics(plan, session, strategy, true);
-                    if (inputTableStatistics.isPresent()) {
-                        Optional<HistoricalPlanStatisticsEntry> historicalPlanStatisticsEntry = getSelectedHistoricalPlanStatisticsEntry(entry.getValue(), inputTableStatistics.get(), historyMatchingThreshold);
-                        if (historicalPlanStatisticsEntry.isPresent()) {
-                            PlanStatistics predictedPlanStatistics = historicalPlanStatisticsEntry.get().getPlanStatistics();
-                            if ((toConfidenceLevel(predictedPlanStatistics.getConfidence()).getConfidenceOrdinal() >= delegateStats.confidenceLevel().getConfidenceOrdinal())) {
-                                return delegateStats.combineStats(
-                                        predictedPlanStatistics,
-                                        new HistoryBasedSourceInfo(entry.getKey().getHash(), inputTableStatistics, Optional.ofNullable(historicalPlanStatisticsEntry.get().getHistoricalPlanStatisticsEntryInfo())));
+                    PlanStatistics predictedPlanStatistics = null;
+                    Optional<HistoricalPlanStatisticsEntry> historicalPlanStatisticsEntry = Optional.empty();
+                    if (getHistoryBasedOptimizerInputStatisticsCheckStrategy(session).equals(NEVER)) {
```

Comment:
If the strategy is to not check the input statistics, we directly use the last history if available.